module/VuFind/src/VuFind/Auth/Shibboleth.php

```diff
@@ -106,22 +155,32 @@ protected function validateConfig()
     public function authenticate($request)
     {
         // Check if username is set.
-        $shib = $this->getConfig()->Shibboleth;
-        $username = $request->getServer()->get($shib->username);
+        $entityId = $this->getCurrentEntityId($request);
+        $shib = $this->getConfigurationLoader()->getConfiguration($entityId);
+        $username = $this->getAttribute($request, $shib['username']);
+        $proxy = $this->getConfig()->Shibboleth->proxy ?? false;
         if (empty($username)) {
+            $details = ($proxy) ? $request->getHeaders()->toArray()
+                : $request->getServer()->toArray();
             $this->debug(
-                "No username attribute ({$shib->username}) present in request: "
-                . print_r($request->getServer()->toArray(), true)
+                "No username attribute ({$shib['username']}) present in request: "
+                . print_r($details, true)
             );
             throw new AuthException('authentication_error_admin');
         }
 
         // Check if required attributes match up:
-        foreach ($this->getRequiredAttributes() as $key => $value) {
-            if (!preg_match('/' . $value . '/', $request->getServer()->get($key))) {
+        foreach ($this->getRequiredAttributes($shib) as $key => $value) {
+            if (!preg_match(
```

Comment:
Write it like this and it'll fit on a single line:

`            if (!preg_match("/$value/", $this->getAttribute($request, $key))) {`